mpisppy/utils/config.py

```diff
@@ -0,0 +1,536 @@
+# This software is distributed under the 3-clause BSD License.
+# Started 12 April by DLW
+# Replace baseparsers.py and enhance functionality.
+# The vision is that global_config ConfigDict can be accessed easily
+# by all code.
+# NOTE: at first I am just copying directly from baseparsers.py
+# TBD: reorganize some of this.
+
+""" Notes 
+The default for all 'with' options is False
+       (and we are dropping the `no` side that was in baseparsers.py)
+
+Now you assemble the args you want and call the create_parser function,
+   which returns an argparse object, E.g.:
+parser = config.create_parser("myprog")
+
+After you you parse with this argparse object (assume it is called parser), 
+you need to call global_config.import_argparse 
+E.g.
+args = parser.parse_args()
+args = global_config.import_argparse(parser)
+
+If you want to add args, you need to call the add_to_config function
+
+If you want a required arg, you have to DYI:
+    parser = config.create_parser("tester")
+    parser.add_argument(
+            "num_scens", help="Number of scenarios", type=int,
+        )
+    args=parser.parse_args(['3', '--max-iterations', '99', '--solver-name', 'cplex'])
+    print(f"{args.num_scens =}")
+(Note: you can still attach it to global_config, but that is also DYI)
+
+"""
+
+"""
+TBD:
+  Do a better job with solver options.
+
+"""
+
+""" Notes on conversion (delete this docstring by summer 2022)
+
+See hydro_cylinders_config.py compared to hydro_cylinders.py
+
+import config (not baseparsers)
+parser = baseparsers.  --> config. (sort of)
+add_argument --> config.add_to_config
+
+Parse_args returns args for the sake of vanilla (which will fail anyway... so vanilla needs to be updated); 
+I want to make a clean break.
+Every use of args gets replaced with  cfg = config.global_config in the main() function
+
+Here's the bummer: args.xxx_yyy becomes cfg["xxx-yyy"], but it's really not too bad 
+"""
+
+import argparse
+import pyomo.common.config as pyofig
+
+global_config = pyofig.ConfigDict()
+
+def _common_args():
+    raise RuntimeError("_common_args is no longer used. See comments at top of config.py")
+
+def popular_args():
+    global_config.declare("max-iterations", pyofig.ConfigValue(
```

Comment:
```suggestion
    global_config.declare("max_iterations", pyofig.ConfigValue(
```

This is better-looking python (e.g., `global_config["max_iterations"]`), and Pyomo config system is smart enough to convert it to `--max-iterations` for the command line.